src/data/MemoryChunk.ts

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+
+export interface Chunk {
+    address: number;
+    size: number;
+}
+
+export default class MemoryChunk implements Chunk {
+    public address: number;
+
+    public hexAddress: number;
+
+    public size: number;
+
+    public consumedSize: number;
+
+    constructor(address: number, size: number, consumedSize: number) {
+        this.address = parseInt(String(address), 10);
+        this.hexAddress = address;
```

Comment:
Both will be equal, js will convert all numbers to decimal for you. this means we don't need to parse `address` as it will already be a decimal number.

If we do parse the string straight from the json file, having a `0x` in front of the number will mess up the `parseInt` with a radix of 10.

Also, can we instead have a getter for `hexAddress` as a string with the added `0x` in front and the number converted to hex?